tests/rptest/services/redpanda.py

```diff
@@ -3286,11 +3305,42 @@ def cloud_storage_client(self):
         return self._cloud_storage_client
 
     def delete_bucket_from_si(self):
-        if self.si_settings.use_bucket_cleanup_policy:
-            self.logger.info(
-                f"Skipping deletion of bucket/container: {self.si_settings.cloud_storage_bucket}."
-                "Using a cleanup policy instead. Please delete it manually")
-            return
+        self.logger.info(
+            f"cloud_storage_cleanup_strategy = {self.si_settings.cloud_storage_cleanup_strategy}"
+        )
+
+        if self.si_settings.cloud_storage_cleanup_strategy == CloudStorageCleanupStrategy.ALWAYS_SMALL_BUCKETS_ONLY:
+            bucket_is_small = True
+            max_object_count = 3000
+
+            # See if the bucket is small enough
+            for i, m in enumerate(
+                    self.cloud_storage_client.list_objects(
```

Comment:
Is this lazy?

Reply:
Yes - very important, right?  I checked, it's always a generator, as one should expect.

Reply:
Indeed, important :).